src/pages/o/[orgId]/surveys/[surveyId]/index.tsx

```diff
@@ -1,32 +1,346 @@
+
+        <RadioGroup
+          name="sig"
+          onChange={(e) => handleRadioChange(e.target.value as SignatureOption)}
+          value={selectedOption}
+        >
+          <RadioFormControlLabel
+            control={<Radio required />}
+            label={
+              <Typography>
+                <Msg
+                  id={messageIds.surveyForm.authenticatedOption}
+                  values={{
+                    email: currentUser?.email ?? '',
+                    person: currentUser?.first_name ?? '',
+                  }}
+                />
+              </Typography>
+            }
+            value="authenticated"
+          />
+
+          <RadioFormControlLabel
+            control={<Radio required />}
+            label={
+              <div>
+                <Typography>
+                  <Msg id={messageIds.surveyForm.nameEmailOption} />
+                </Typography>
+              </div>
+            }
+            value="email"
+          />
+
+          {selectedOption === 'email' && (
+            <Box display="flex" flexDirection="column">
+              <TextField label="First Name" name="sig.first_name" required />
+              <TextField label="Last Name" name="sig.last_name" required />
+              <TextField label="Email" name="sig.email" required />
+            </Box>
+          )}
+
+          {survey.signature === 'allow_anonymous' && (
+            <RadioFormControlLabel
+              control={<Radio required />}
+              label={
+                <Typography>
+                  <Msg id={messageIds.surveyForm.anonymousOption} />
+                </Typography>
+              }
+              value="anonymous"
+            />
+          )}
+        </RadioGroup>
+
+        <Box alignItems="center" component="section" sx={{ py: 2 }}>
+          <Typography fontWeight={'bold'}>
+            <Msg id={messageIds.surveyForm.terms.title} />
+          </Typography>
+
+          <FormControlLabel
+            control={<Checkbox required />}
+            data-testid="Survey-acceptTerms"
+            label={<Msg id={messageIds.surveyForm.accept} />}
+            name="privacy.approval"
+          />
+          <Typography style={{ fontSize: '0.8em' }}>
+            <Msg
+              id={messageIds.surveyForm.terms.description}
+              values={{ organization: survey.organization.title }}
+            />
+          </Typography>
+          <Typography style={{ fontSize: '0.8em', marginBottom: '0.5em' }}>
+            <Link
+              href={messages.surveyForm.policy.link()}
+              rel="noreferrer"
+              target="_blank"
+            >
+              <Msg id={messageIds.surveyForm.policy.text} />
+            </Link>
+          </Typography>
+        </Box>
+
+        <Button
+          color="primary"
+          data-testid="Survey-submit"
+          style={{ textAlign: 'center', width: '100%' }}
+          type="submit"
+          variant="contained"
+        >
```

Comment:
This should be disabled unless the user has checked the terms checkbox (unless you know of an accessibility or other reason not to do it that way?).

Reply:
The `required` attribute on the terms checkbox takes care of this beautifully. Check it out, here's a quick GIF that shows visually what VoiceOver announces when you try to submit the form without accepting the terms.

<img width="333" src="https://github.com/zetkin/app.zetkin.org/assets/566159/3d329213-c3eb-41aa-add9-9bfd80c7cfaa" />

On submit, it announces this.

> I accept the terms stated below, required invalid data unticked, Tick box.
> Please tick this box if you want to proceed.

From an accessibility POV this is pretty great because it means there's no mystery for the user about which specific input field is preventing them from submitting the form.

The alternatives here are to stick `disabled` on the submit button, or to put `aria-disabled` on it and call `preventDefault` in submit handler. Using `disabled` means the button is effectively the same as `aria-hidden` from a screen reader POV though, so while a visual user may see the button and understand "oh I can't click yet maybe I screwed up", it raises the technical bar a bit for visually impaired users. Using `aria-disabled` avoids hiding the button completely: a screen reader user will still be able to perceive the button's presence and it'll be announced as "disabled", but then you gotta write extra code to call `preventDefault` and actually prevent submitting the form on click, and you gotta handle surfacing the reason for the blockage yourself too.

Think this might actually be another one of those cases where less is more and using the platform is the best bet.

Reply:
Love it! I really appreciate your knowledge and experience on this. Of course we should go with your solution then. Thanks for explaining it!